File: Languages/Java/ReverseLinkedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    static ListNode revH = null;
    public ListNode reverseList(ListNode head) {
       if(head==null)
           return null;
        revLL(head);
        return revH;
    }
    
    public  void revLL(ListNode head){
        
         if(head.next == null){
            revH = head;
             return;
           }
        
        revLL(head.next);
        head.next.next = head;
        head.next = null;
       
    }
    
}

File: C++/Top 180 Interview questions/solutions/reverselinkedlist.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *temphead=NULL;
        while(head!=NULL)
        {
            ListNode* temp= head->next;
            head->next=temphead;
            temphead=head;
            head=temp;
        }
        return temphead;
    }
};